Thank you for your contribution @drewnoakes 
Some tests are failed to compile, please check.
```
C:\Test\NuGet.Client\test\NuGet.Clients.Tests\NuGet.VisualStudio.Implementation.Test\SolutionExplorer\Models\Asset
sFileDependenciesSnapshotTests.cs(62,63): error CS7036: There is no argument given that corresponds to the required par
ameter 'lockFileTarget' of 'AssetsFileDependenciesSnapshot.ParseLibraries(LockFile, LockFileTarget)' [C:\Test\NuGe
t.Client\test\NuGet.Clients.Tests\NuGet.VisualStudio.Implementation.Test\NuGet.VisualStudio.Implementation.Test.csproj:
:TargetFramework=net472]
C:\Test\NuGet.Client\test\NuGet.Clients.Tests\NuGet.VisualStudio.Implementation.Test\SolutionExplorer\Models\Asset
sFileDependenciesSnapshotTests.cs(89,99): error CS7036: There is no argument given that corresponds to the required par
ameter 'lockFileTarget' of 'AssetsFileDependenciesSnapshot.ParseLibraries(LockFile, LockFileTarget)' [C:\Test\NuGe
t.Client\test\NuGet.Clients.Tests\NuGet.VisualStudio.Implementation.Test\NuGet.VisualStudio.Implementation.Test.csproj:
:TargetFramework=net472]
C:\Test\NuGet.Client\test\NuGet.Clients.Tests\NuGet.VisualStudio.Implementation.Test\SolutionExplorer\Models\Asset
sFileDependenciesSnapshotTests.cs(127,112): error CS7036: There is no argument given that corresponds to the required p
arameter 'lockFileTarget' of 'AssetsFileDependenciesSnapshot.ParseLibraries(LockFile, LockFileTarget)' [C:\Test\Nu
Get.Client\test\NuGet.Clients.Tests\NuGet.VisualStudio.Implementation.Test\NuGet.VisualStudio.Implementation.Test.cspro
j::TargetFramework=net472]
```

--- a/test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/SolutionExplorer/Models/AssetsFileDependenciesSnapshotTests.cs
+++ b/test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/SolutionExplorer/Models/AssetsFileDependenciesSnapshotTests.cs
@@ -59,7 +59,7 @@ namespace NuGet.VisualStudio.Implementation.Test.SolutionExplorer.Models
             var lockFileFormat = new LockFileFormat();
             var lockFile = lockFileFormat.Parse(lockFileContent, "In Memory");
 
-            var dependencies = AssetsFileDependenciesSnapshot.ParseLibraries(lockFile.Targets.First());
+            var dependencies = AssetsFileDependenciesSnapshot.ParseLibraries(lockFile, lockFile.Targets.First());
 
             Assert.Equal(1, dependencies.Count);
             Assert.True(dependencies.ContainsKey("system.runtime"));
@@ -86,7 +86,7 @@ namespace NuGet.VisualStudio.Implementation.Test.SolutionExplorer.Models
                 }
             };
 
-            var exception = Assert.Throws<ArgumentException>(() => AssetsFileDependenciesSnapshot.ParseLibraries(lockFileTarget));
+            var exception = Assert.Throws<ArgumentException>(() => AssetsFileDependenciesSnapshot.ParseLibraries(new LockFile(), lockFileTarget));
 
             Assert.Contains("PackageA", exception.Message);
         }
@@ -124,7 +124,7 @@ namespace NuGet.VisualStudio.Implementation.Test.SolutionExplorer.Models
                 }
             };
 
-            ImmutableDictionary<string, AssetsFileTargetLibrary> dependencies = AssetsFileDependenciesSnapshot.ParseLibraries(lockFileTarget);
+            ImmutableDictionary<string, AssetsFileTargetLibrary> dependencies = AssetsFileDependenciesSnapshot.ParseLibraries(new LockFile(), lockFileTarget);
 
             Assert.Equal(lockFileTarget.Libraries.Count, dependencies.Count);
             Assert.All<LockFileTargetLibrary>(lockFileTarget.Libraries,

